amgcl/backend/builtin.hpp

```diff
@@ -605,6 +606,44 @@ struct inner_product_impl<
 };
 
 template < class Vec1, class Vec2 >
+struct inner_product_impl<
+    Vec1, Vec2,
+    typename boost::enable_if<
+            typename boost::mpl::and_<
+                typename is_builtin_vector<Vec1>::type,
+                typename is_builtin_vector<Vec2>::type,
+                typename boost::is_complex<typename value_type<Vec1>::type>::type
+                >::type
+        >::type
+    >
+{
+    typedef typename value_type<Vec1>::type V;
+
+    static V get(const Vec1 &x, const Vec2 &y)
+    {
+        const size_t n = x.size();
+        V sum = 0;
+
+        {
+            size_t chunk_start = 0;
+            size_t chunk_end   = n;
```

Comment:
This is leftover code from openmp implementation of the generic inner product. We can either remove the code and keep the loop below serial, or we can do something like
```cpp
#pragma omp parallel
{
    ...
    for(...) {
        ...
    }
#pragma omp atomic
    sum += s;
}
return sum;
```